waspc/data/Generator/templates/server/src/auth/providers/oauth/createRouter.ts

```diff
@@ -42,48 +49,53 @@ export function createRouter(provider: ProviderConfig, initData: { passportStrat
               throw new Error(`${provider.displayName} provider profile was missing required id property. This should not happen! Please contact Wasp.`);
           }
 
-          // Wrap call to getUserFieldsFn so we can invoke only if needed.
-          const getUserFields = () => getUserFieldsFn(contextWithUserEntity, { profile: providerProfile });
-          // TODO: In the future we could make this configurable, possibly associating an external account
-          // with the currently logged in account, or by some DB lookup.
-          const user = await findOrCreateUserByExternalAuthAssociation(provider.id, providerProfile.id, getUserFields);
+          const providerId = createProviderId(provider.id, providerProfile.id);
 
-          const token = await sign(user.id);
-          res.json({ token });
+          try {
+            // We need a user id to create the auth token, so we either find an existing user
+            // or create a new one if none exists for this provider.
+            const getUserIdFromProviderDetails = async () => {
```

Comment:
I did not want to separate it into two functions since it then felt needlessly chopped up.

The meaning of the code is "get a user id based on provider details" it can be an existing user or it can be a previously unseen user (which gets created) and I think it makes sense to keep it in one place. I've now extracted the code into a separate function to make it more readable for you. The inline style was okay for me since it allowed the function to pick up the variables available in the scope, but I get your point of there being a lot of code.

Reply:
The ability for it to pick up the variables in the scope is exactly the problem: it is hard to say what it is interacting with. It wouldn't be a problem if the whole code was smaller, but this is a quite long piece of code and so it becomes hard to follow what is or is not being used.
Feel free to resolve this comment thread once you read it!